core/storage/file/gae_models_test.py

```diff
@@ -0,0 +1,229 @@
+        with self.assertRaisesRegexp(
+            base_models.BaseModel.EntityNotFoundError, (
+                'Entity for class FileMetadataModel '
+                'with id /exp_id1/path/to/file2.png not found')):
+            file_models.FileMetadataModel.get_model(
+                'exp_id1', 'path/to/file2.png', True)
+
+    def test_get_version_with_version_present_returns_correct_model(self):
+        file_metadata_model = file_models.FileMetadataModel.create(
+            'exp_id1', 'path/to/file1.png')
+        file_metadata_model.commit(feconf.SYSTEM_COMMITTER_ID, [])
+
+        retrieved_model = file_models.FileMetadataModel.get_version(
+            'exp_id1', 'path/to/file1.png', 1)
+        self.assertEqual(file_metadata_model.key, retrieved_model.key)
+
+        file_metadata_model.commit(feconf.SYSTEM_COMMITTER_ID, [])
+        retrieved_model = file_models.FileMetadataModel.get_version(
+            'exp_id1', 'path/to/file1.png', 2)
+        self.assertEqual(file_metadata_model.key, retrieved_model.key)
+
+    def test_get_version_with_version_absent_raises_error(self):
+        file_metadata_model = file_models.FileMetadataModel.create(
+            'exp_id1', 'path/to/file1.png')
+        file_metadata_model.commit(feconf.SYSTEM_COMMITTER_ID, [])
+
+        with self.assertRaisesRegexp(
+            base_models.BaseModel.EntityNotFoundError, (
+                'Entity for class FileMetadataSnapshotContentModel with id '
+                '/exp_id1/path/to/file1.png-2 not found')):
+            file_models.FileMetadataModel.get_version(
+                'exp_id1', 'path/to/file1.png', 2)
+
+    def test_commit_updates_version_of_stored_model(self):
+        file_metadata_model = file_models.FileMetadataModel.create(
+            'exp_id1', 'path/to/file1.png')
+        self.assertEqual(file_metadata_model.version, 0)
+
+        file_metadata_model.commit(feconf.SYSTEM_COMMITTER_ID, [])
+        self.assertEqual(file_metadata_model.version, 1)
+
+
+class FileModelTest(test_utils.GenericTestBase):
+    """Tests the FileModel class."""
+
+    def test_file_model_content_is_reconstituted_correctly(self):
+        file_model = file_models.FileModel.create(
+            'exp_id1', 'path/to/file1.png')
+        file_model.commit(feconf.SYSTEM_COMMITTER_ID, [])
+
+        file_model.content = 'file_contents'
+        commit_cmds = [{'cmd': 'edit'}]
+        file_model.commit(feconf.SYSTEM_COMMITTER_ID, commit_cmds)
+        retrieved_model = file_models.FileModel.get_version(
+            'exp_id1', 'path/to/file1.png', 2)
+        self.assertEqual(file_model.key, retrieved_model.key)
+        self.assertEqual(retrieved_model.content, 'file_contents')
+
+    def test_file_model_snapshot_includes_file_model_content(self):
+        file_model = file_models.FileModel.create(
+            'exp_id1', 'path/to/file1.png')
+
+        self.assertIsNone(file_model.content)
```

Comment:
You could move this bit to its own test, e.g. something along the lines of:

``test_initial_file_model_has_no_content``

This helps encourage testing only one behavior per test method.

Ditto for other tests that have the observable pattern:

- Arrange
- Act 1
- Assert 1
- Act 2
- Assert 2

These can generally be split into two (or more) tests which instead follow

- Arrange
- Act 1
- Assert 1

- Arrange + Act 1 (where 'act 1' now represents part of the arrangement)
- Act 2
- Assert 2

Where each test is verifying a related, but slightly similar, behavior.